common/app/targeting/CampaignAgent.scala

```diff
@@ -0,0 +1,43 @@
+package targeting
+
+import common._
+import com.gu.targeting.client._
+import com.amazonaws.services.s3.AmazonS3Client
+import conf.Configuration
+import services.AwsEndpoints
+
+import scala.util.Try
+import scala.util.control.NonFatal
+
+object CampaignAgent extends Logging with ExecutionContexts {
+  val crossAccountClient: Option[AmazonS3Client] = Try({
+    val client = new AmazonS3Client(Configuration.targeting.crossAccountMandatoryCredentials)
+    client.setEndpoint(AwsEndpoints.s3)
+    client
+  }).toOption
+
+  private val agent = AkkaAgent[CampaignCache](new CampaignCache())
+
+  def refresh() {
+    crossAccountClient.foreach(client => {
+      agent.alter { old =>
+        try {
+          old.update(client, Configuration.targeting.campaignsBucket)
+        } catch {
+          case NonFatal(e) => log.error("Failed to update campaign list.", e)
+        }
+        old
```

Comment:
Why returning old? Since you don't seem to do anything with the resulting `Future[CampaignCache]` you might want to use `send` rather than `alter`